mu-plugins/blocks/table-of-contents/src/view.js

```diff
@@ -67,12 +67,19 @@ function onScroll() {
 
 function init() {
 	const container = document.querySelector( '.wp-block-wporg-table-of-contents' );
+	// Margin offset from the top of the sidebar.
+	const gap = getCustomPropValue( '--wp--preset--spacing--edge-space' );
 
 	if ( container ) {
+		// Usable viewport height.
 		const viewHeight = window.innerHeight - FIXED_HEADER_HEIGHT;
+		// Get the height of the sidebar, plus the top margin and 50px for the
+		// "Back to top" link, which isn't visible until `is-fixed-sidebar` is
+		// added, therefore not included in the parentNode.offsetHeight value.
+		const sidebarHeight = container.parentNode?.offsetHeight + gap + 50;
```

Comment:
Change to the font size and the margin of the "Back to Top" link could throw this off, but I think this is an immediate improvement.